tests/BlazingSQLTest/EndToEndTests/TestSuites/intervalSuite.yaml

```diff
@@ -0,0 +1,302 @@
+            from orders order by o_totalprice limit 100
+
+  TEST_42:
+    SQL: |
+        select o_orderdate, o_orderdate - INTERVAL '2' SECOND as sub_sec_col
+            from orders order by o_totalprice limit 150
+
+  TEST_43:
+    SQL: |
+        select o_orderdate, o_orderdate + INTERVAL '6' MINUTE as add_min_col
+            from orders order by o_totalprice limit 50
+
+  TEST_44:
+    SQL: |
+        select o_orderdate, o_orderdate - INTERVAL '5' MINUTE as sub_min_col
+            from orders order by o_totalprice limit 50
+
+  TEST_45:
+    SQL: |
+        select o_orderdate, o_orderdate + INTERVAL '1' HOUR as add_hour_col
+            from orders order by o_totalprice limit 50
+
+  TEST_46:
+    SQL: |
+        select o_orderdate, o_orderdate - INTERVAL '4' HOUR as sub_hour_col
+            from orders order by o_totalprice limit 85
+        
+  TEST_47:
+    SQL: |
+         select o_orderdate, o_orderdate + INTERVAL '1' DAY as add_day_col
+            from orders order by o_totalprice limit 100
+
+  TEST_48:
+    SQL: |
+        select o_orderdate, o_orderdate - INTERVAL '7' DAY as sub_day_col
+            from orders order by o_totalprice limit 50
+
+
+  # ========================= DURATION op INTERVAL ============================
+
+  TEST_49:
+    SQL: |
+        select * from interval_table
+
+  TEST_50:
+    SQL: |
+        select i_duration_s, i_duration_ms from interval_table
+
+  TEST_51:
+    SQL: |
+        select i_id, i_duration_s from interval_table
+              order by i_duration_s desc limit 25
+
+  # Join with DURATION same unit
+  TEST_52:
+    SQL: |
+        select r1.i_id + 1 as r1_id, r1.i_duration_s as r1_duration_s,
+                r2.i_duration_ms as r2_duration_ms
+            from interval_table r1 inner join interval_table r2
+            on r1.i_duration_s = r2.i_duration_s
+            order by r1.i_id limit 70       
+
+  # Join with DURATION same unit
+  TEST_53:
+    SQL: |
+        select r1.i_id as r1_id, r1.i_duration_s as r1_duration_s,
+                r1.i_duration_ms as r1_duration_ms
+            from interval_table r1 inner join interval_table r2
+            on r1.i_duration_ms = r2.i_duration_s
+
+  TEST_54:
+    SQL: |
+        select count(i_duration_s) from interval_table
+
+  TEST_55:
+    SQL: |
+        select sum(i_id) as sum_col, i_duration_s from interval_table
+            group by i_duration_s
+
+  # union with DURARION same unit
+  TEST_56:
+    SQL: |
+        (select i_id, i_duration_ms from interval_table where i_id > 236)
+            union all 
+        (select i_id, i_duration_ms from interval_table where i_id < 55)
+
+  # union with DURARION diff unit
+  TEST_57:
+    SQL: |
+        (select i_id, i_duration_s from interval_table limit 36)
+            union all 
+        (select i_id, i_duration_ms from interval_table limit 4)
+
+
+  # NOTE: all this query works for Blazing but not for Drill
+  # TEST_58
+  #   select i_duration_s, INTERVAL '1' SECOND as col_to_sum,
```

Comment:
do these work for Spark?

Reply:
All these tests were just faced again Drill due to the schema. Spark does not support `timedelta64[X]` variations as Drill does.